docs/notebooks/how_to_build_and_optimize_qaoa_using_swap_strategy.ipynb

```diff
@@ -0,0 +1,742 @@
+      "text/plain": [
+       "<Figure size 2210.55x2709 with 1 Axes>"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from qiskit.circuit.library import QAOAAnsatz\n",
+    "from qiskit.quantum_info import SparsePauliOp\n",
+    "\n",
+    "cost_operator = SparsePauliOp.from_list(cost_hamiltonian)\n",
+    "ansatz = QAOAAnsatz(\n",
+    "    cost_operator,\n",
+    "    reps=2,\n",
+    ").decompose()\n",
+    "ansatz.decompose(reps=2).draw(output=\"mpl\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.2 Optimizing the quantum circuit for quantum execution using the SWAP strategy and the SAT mapping\n",
+    "SWAP gate insertion is the technique of choice for most quantum circuits to run on quantum hardware with limited qubit connectivity.  The first couple of steps of the workflow depend on the choice of qubit configuration we want, which will determine the kind of SWAP strategy that will be used, as well as the evaluation criteria for the backend to obtain the best subset of qubits.\n",
+    "\n",
+    "In this example, we choose a swap insertion strategy for blocks of commuting two-qubit gates which applies layers of SWAP gates simultaneously executable on the coupling map. This strategy is presented in [Weidenfeller et al.](https://arxiv.org/abs/2202.03459) and exposed as a standardized Qiskit transpiler pass, where we can choose different qubit configurations, in this case, a line."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.transpiler.passes.routing.commuting_2q_gate_routing import SwapStrategy\n",
+    "\n",
+    "swap_strategy = SwapStrategy.from_line([i for i in range(num_qubits)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Remap the graph using a SAT mapper\n",
+    "\n",
+    "Even when a circuit consists of commuting gates (this is the case for the QAOA circuit, but also for trotterized simulations of Ising Hamiltonians), finding a good initial mapping is a hard problem. The SAT-based approach presented in [Matsuo et al.](https://arxiv.org/abs/2212.05666) allows to find good initial mappings for circuits with commuting gates, which lead to a reduced number of swap layers.\n",
+    "\n",
+    "The [`qopt_best_practices` repository](https://github.com/qiskit-community/qopt-best-practices)  exposes the mapper class from Matsuo et al. (`SATMapper`) and includes the `remap_graph_with_sat` method, enabling the remapping of the problem to a more optimal initial state for a specified swap strategy. This method can significantly reduce the number of swap layers required to execute the circuit.\""
```

Comment:
You introduced this module before without referencing. It might be better to reference the first time you mention it 🙂